pkg/provider/common.go

```diff
@@ -0,0 +1,28 @@
+package provider
+
+import (
+	"context"
+	"os"
+
+	"github.com/giantswarm/microerror"
+	capi "sigs.k8s.io/cluster-api/api/v1alpha3"
+	ctrl "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	Provider = "E2E_PROVIDER"
+)
+
+func GetProviderSupport(ctx context.Context, client ctrl.Client, cluster *capi.Cluster) (Support, error) {
+	switch os.Getenv(Provider) {
+	case "azure":
+		p, err := NewAzureProviderSupport(ctx, client, cluster)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+
+		return p, nil
+	}
+
+	return nil, microerror.Maskf(executionFailedError, "unsupported provider")
```

Comment:
just a nitpick. Maybe worth mentioning either the supplied env variable value (to catch for trim-like errors) and/or the supported providers?

Reply:
I'll do both. I'll trim the env value, but also improve the error. :+1: